src/material-experimental/mdc-radio/radio.scss

```diff
@@ -86,6 +86,20 @@
   }
 }
 
+// This is necessary because we do not depend on MDC's ripple, but have our own that should be
+// positioned correctly. This can be removed once we start using MDC's ripple implementation.
+.mat-mdc-radio-checked .mdc-radio {
```

Comment:
Which ripple is this fixing exactly? As far as I can tell, the focus ripple already has the correct colors. If it's supposed to fix the click ripple, you can do it simpler by adding the following rule under the `&.mat-mdc-radio-checked` selector:

```
.mat-ripple-element {
  background-color: var(--mat-mdc-radio-checked-ripple-color, transparent);
}
```